localstack-core/localstack/services/apigateway/next_gen/execute_api/handlers/gateway_exception.py

```diff
@@ -0,0 +1,87 @@
+import json
+import logging
+
+from rolo import Response
+
+from localstack.aws.api.apigateway import GatewayResponse, GatewayResponseType
+from localstack.services.apigateway.next_gen.execute_api.api import (
+    RestApiGatewayExceptionHandler,
+    RestApiGatewayHandlerChain,
+)
+from localstack.services.apigateway.next_gen.execute_api.context import RestApiInvocationContext
+from localstack.services.apigateway.next_gen.execute_api.gateway_response import (
+    BaseGatewayException,
+    Default5xxError,
+    build_default_response,
+)
+
+LOG = logging.getLogger(__name__)
+
+
+class GatewayExceptionHandler(RestApiGatewayExceptionHandler):
+    """
+    Exception handler that serializes the Gateway Exceptions into Gateway Responses
+    """
+
+    def __call__(
+        self,
+        chain: RestApiGatewayHandlerChain,
+        exception: Exception,
+        context: RestApiInvocationContext,
+        response: Response,
+    ):
+        if not isinstance(exception, BaseGatewayException):
+            (
+                LOG.warning(
+                    "Non Gateway Exception raised: %s",
+                    exception,
+                    exc_info=LOG.isEnabledFor(logging.DEBUG),
+                ),
+            )
+            exception = Default5xxError(message=f"LocalStack Error: {exception}", status_code="500")
+
+        error = self.create_exception_response(exception, context)
+        if error:
+            response.update_from(error)
+
+    def create_exception_response(
+        self, exception: BaseGatewayException, context: RestApiInvocationContext
+    ):
+        gateway_response = self._get_gateway_response(exception, context)
+
+        content = self._build_response_content(exception)
+
+        headers = self._build_response_headers()
+
+        status_code = gateway_response.get("statusCode", exception.status_code)
```

Comment:
I think this will work fine as the user defined one would be part of the `gateway_response` and each exception carries it's own default if the user provided no override.